### DISCUSSION TITLE
Issues with phalcon dev tools

### DISCUSSION THREAD

**Discussion by krishnan-ubuntu** [phalcon/cphalcon#15513] [Q&A] [answered]
I'm new to phalcon and I'm having issues with phalcon dev tool commands.

When I type phalcon serve I get the following error.

```
Phalcon DevTools (4.1.2)

PHP Fatal error:  Class 'Phalcon\Mvc\User\Component' not found in /opt/phalcon-devtools/scripts/Phalcon/Utils/SystemInfo.php on line 36
```
I looked for answers but couldn't find much. Thanks in advance.

**Comment by quasipickle** [comment, marked as answer]
Back in v3, components would extend `Phalcon\Mvc\User\Component`.  In v4, that has been changed to `\Phalcon\Di\Injectable` - which you can see here: https://github.com/phalcon/phalcon-devtools/blob/ad7a6eb1a109f62ff07d6eb5abf6c370fdbeaa2b/src/Utils/SystemInfo.php

This error is seeming to say that `/opt/phalcon-devtools/scripts/Phalcon/Utils/SystemInfo.php` is a version of the file from 3.4, since it's trying to extend `\Phalcon\Mvc\User\Component`.  My guess is the version of the devtools you're using is 4.1.2, and the version of Phalcon you have installed is 4.1.2, but the scripts that got installed into `/opt/phalcon-devtools/scripts/` are still the 3.4 version.

I'd try removing what's there and reinstalling.

```shell
composer global remove phalcon/devtools
composer global require phalcon/devtools
```
